### ISSUE TITLE
Set Focus

### ISSUE THREAD

**Issue by robodna** [PolymerElements/paper-input#471]
### Description

Calling .focus() on paper-input in Shady Dom does not correctly set focus, but does in Shadow Dom.  A gray border appears and focus is not set.  The autofocus attribute has no affect and this may be related to the issue.
### Expected outcome
The paper-input's bottom border should animate and increase in size to indicate focus has been obtained.  

### Actual outcome
A gray box appear around the paper-input when .focus() is called, and the focus is not set.

### Steps to reproduce
Call .focus() on a paper-input in Shady Dom, or set the autofocus attribute on a paper-input in Shady Dom.

### Browsers Affected

- [x] Chrome
- [ ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

**Comment by jimbojetlag** [comment]
Also, `autofocus` on paper-input does not work with shady dom.

**Comment by jimbojetlag** [comment]
Also, `autofocus` on paper-input does not work with shady dom.

**Comment by notwaldorf** [comment]
I can't repro this: http://output.jsbin.com/nixoqim

This is the result of calling `i1.focus()`
<img width="254" alt="screen shot 2017-01-04 at 3 59 44" src="https://cloud.githubusercontent.com/assets/1369170/21663725/d525fb62-d296-11e6-84ae-d07ba89dbecf.png">

Closing, but please reopen if you have reproducible steps.

**Comment by jimbojetlag** [comment]
@notwaldorf your paper-input is not inside a template tag, the problem happens when paper-input is in a template. The following works in Chrome, but not in Firefox:

http://plnkr.co/edit/mQnovfFtoF35SZM9ntzA?p=preview

It's supposed to auto focus.

**Comment by notwaldorf** [comment]
That detail was not at all present in your issue description :)

**Comment by notwaldorf** [comment]
TBH that sounds like it's a Shadow DOM bug and not an easy fix -- I don't think the `autofocus` attribute is visible, since the `paper-input` is in the shadow DOM.

**Comment by jimbojetlag** [comment]
Shadow DOM is still buggy? I thought it was supposed to be production ready after so many years.

Which Shadow DOM bug is this related to? Is there an open issue for it?

**Comment by jimbojetlag** [comment]
> That detail was not at all present in your issue description :)

I didn't open this issue :)

**Comment by robodna** [comment]
The issue I opened was resolved on my end, but I'm not certain what what 
causing it.

I would call paper-input.focus() once when I switched iron-pages to the 
edit view, a second time when an API response was received to populate 
the edit form, and then a third time when the iron-overlay closed.  
Having the first and second only resulted in focus not being set 
correctly.  Adding the third .focus() call when the overlay closed only 
worked after I removed the first and second .focus() calls.  There seems 
to have been some kind of race condition or conflict with multiple 
.focus() calls.

On 17-01-07 12:17 PM, jimbojetlag wrote:
>> That detail was not at all present in your issue description :)
> I didn't open this issue :)
>

-- 
Lou Morris
RoboDNA.com

**Comment by notwaldorf** [comment]
@jimbojetlag Ah sorry, I haven't actually tested the `autofocus` attribute with shadow DOM -- I was just offering an explanation why it might not work, if in fact it did not work.

@robodna Should I close this issue? Is it resolved, then?

**Comment by robodna** [comment]
Yes, close it, I'm not able to duplicate the issue other than re-adding 
the mutiple .focus() calls.

On 17-01-09 01:31 PM, Monica Dinculescu wrote:
> @jimbojetlag Ah sorry, I haven't actually tested the `autofocus` attribute with shadow DOM -- I was just offering an explanation why it might not work, if in fact it did not work.
>
> @robodna Should I close this issue? Is it resolved, then?
>

-- 
Lou Morris
RoboDNA.com
